src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -50,30 +52,35 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
         if (!$options->class || $options->disabled) {
             return [];
         }
-        if (!$manager = $this->getManager($options->objectManager, $options->class)) {
+
+        $class = \array_key_exists($options->class, $this->typeAliases)
+            ? $this->typeAliases[$options->class]
+            : $options->class;
```

Comment:
let's mutate the attribute and remove most of the changes on this PR?
```suggestion
        if (isset($this->typeAliases[$options->class])) {
           $options->class = $this->typeAliases[$options->class];
        }
```

Reply:
I considered this, but figured that might cause issues with other potential consumers of this attribute, so keeping a local state seemed like a cleaner solution.

Alternatively this could be completely extracted to the Doctrine bundle by wrapping this ValueResolver over there, like in https://github.com/symfony/symfony/issues/51765#issuecomment-2044828302; this means we can mutate the property before calling the resolver, and afterwards restore it to the original value.

Looking back I'm not sure which approach I think is better; considering the relevant configuration lives in the bundle as well, and the changes required to the DoctrineExtension class in the bundle to wire this parameter in the bridge, it might be cleaner to fully encapsulate it in just the bundle. What do you think?

Reply:
This should definitely be implemented here. We're in control of both the bridge and the bundle. The design principle is that the bundle should be only about wiring stuff from the bridge.

About the side effect, let's use this then?

```php
$options = (new MapEntity($this->typeAliases[$options->class] ?? null))->withDefaults($options);
```

Reply:
Thanks for the feedback; it is definitely better placed here then.

Creating a new MapEntity instance with the translated value seems like a good solution to me 👍

Reply:
I ended up taking your initial suggestion anyway, as the withDefaults call done in the original code already creates a clone of the original MapEntity, which means no side effects should occur.